### PR TITLE
Fix tests hub failure

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

This PR fixes the flaky tests in the test_hub job. 

TL;DR: the two tests changed push to the Hub with async commits which is the source of two flaky errors and one at-exit error:
- 1st flaky error: the pushes may not be finished before we try to access the repo, so we wait for the push in progress to be finished
- 2nd flaky error: the pushes may not all be done, since a commit is skipped when the push in progress is not finished

At-exit error: when exiting the test, `huggingface_hub` accesses the `is_done` property of some of the commands (even if they are finished), which calls another time the `post_method` of those command (`lfs_prune`) on a repo that does not exist anymore. This is fixed on the `huggingface_hub` side by the PR mentioned above.

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
Good solution, thanks for working on it @sgugger!
